Add security checklist (#2141)
alan-turing-institute/data-safe-haven: docs/source/deployment/security_checklist.md

Do not register this user with any SRE yet.

## 1. Multifactor authentication and password strength

### Turing configuration setting:

- Users must set up MFA before accessing the secure analysis environment.
- Users cannot access the environment without MFA.
- Users are required/advised to create passwords of a certain strength.

### Implication:

- Users are required to authenticate with Multi-factor Authentication (MFA) in order to access the secure analysis environment.
- Passwords are strong

### Verify by:

#### Check: Users can reset their own password

- Attempt to login to the remote desktop web client as the research user.
- Click "Forgotten my password".
- Reset password.

````{attention}
{{camera}} <b>Verify that:</b>
 <details><summary> user can reset their own password</summary>

```{image} security_checklist/sspr.png
:align: center
```
```{image} security_checklist/sspr_success.png
:align: center
```

</details>
````

#### Check: Non-registered users cannot connect to any SRE workspace

Attempt to login to the remote desktop web client as the research user.

````{attention}
{{camera}} <b>Verify that:</b>

>  <details><summary>user can authenticated but cannot see any workspaces</summary>

Reviewer: ```suggestion
 <details><summary>user can authenticate but cannot see any workspaces</summary>
```

```{image} security_checklist/no_valid_workspaces.png
:align: center
```

</details>
````

#### Check: Registered users can see SRE workspaces

Check that the research user can authenticate using MFA and is granted access to the SRE.

- Login to the remote desktop web client as the research user.

````{attention}
{{camera}} <b>Verify that:</b>

> <details><summary>user can authenicated and can see workspaces</summary>

Reviewer: ```suggestion
<details><summary>user can authenticate and can see workspaces</summary>
```

```{image} security_checklist/valid_workspaces.png
:align: center
```

</details>
````

#### Check: Authenticated user can access workspaces

Check that the research user can access a workspace.

- Login to the remote desktop web client as the research user.
- Select a workspace and login as the research user.

````{attention}
{{camera}} <b>Verify that:</b>
<details><summary>you can connect to any workspace</summary>

```{image} security_checklist/workspace_xfce_initial.png
:align: center
```

</details>
````

## 2. Isolated Network

### Turing configuration setting:

> - The only part of the SRE a {ref}`Researchers <role_researcher>` can access from the internet is the remote desktop web client.

Reviewer: ```suggestion
- The only part of the SRE a {ref}`Researcher <role_researcher>` can access from the internet is the remote desktop web client.
```

> - From within the SRE, a {ref}`Researchers <role_researcher>` cannot use the connect to clients outside the SRE network (with the exception of indirect, read-only access to package repositories).

Reviewer: ```suggestion
- From within the SRE, a {ref}`Researcher <role_researcher>` cannot connect to clients outside the SRE network (with the exception of indirect, read-only access to package repositories).
```

- Use the IP address of your own device in place of that of the data provider.
- Generate an upload token with only Write and List permissions.

```{attention}
{{white_check_mark}} Verify that: the upload token is successfully created.
```

```{attention}
{{white_check_mark}} Verify that: you are able to send this token using a secure mechanism.
```

#### Ensure that data ingress works only for connections from the accepted IP address range

- As the {ref}`role_data_provider_representative`, ensure you're working from a device that has an allowed IP address.
- Using the upload token with write-only permissions and limited time period that you set up in the previous step, follow the ingress instructions for the {ref}`data provider <role_data_provider_representative>`.

```{attention}
{{white_check_mark}} Verify that: writing succeeds by uploading a file
```

```{attention}
{{white_check_mark}} Verify that: attempting to open or download any of the files results in the following error: `Failed to start transfer: Insufficient credentials.` under the `Activities` pane at the bottom of the MS Azure Storage Explorer window.
```

> - Switch to a device without and allowed IP address (or change your IP with a VPN)

Reviewer: ```suggestion
- Switch to a device without an allowed IP address (or change your IP with a VPN)
```

```

#### Confirm that {ref}`role_system_manager` can see and download files from output

- As the {ref}`role_system_manager`, follow the instructions in the [project manager documentation](../roles/project_manager/data_egress.md#data-egress-process) on how to access files set for egress with `Azure Storage Explorer`.

```{attention}
{{white_check_mark}} Verify that: you can see the files written to the `/output` storage volume.
```

```{attention}
{{white_check_mark}} Verify that: a written file can be taken out of the environment via download
```

## 9. Software package repositories

### Turing configuration setting::

- {ref}`policy_tier_2`: The user can access any package from our mirrors or via our proxies. They can freely use these packages without restriction.
- {ref}`policy_tier_3`: The user can only access a specific pre-agreed set of packages. They will be unable to download any package not on the allowed list.

### Implication:

- {ref}`policy_tier_2`: User can access all packages from PyPI/CRAN

> - {ref}`policy_tier_3`: User can only access approved packages from PyPI/CRAN. Allowed list is in `environment_configs/package_lists`

Reviewer: ```suggestion
- {ref}`policy_tier_3`: User can only access approved packages from PyPI/CRAN. Allowed lists are in `data_safe_haven/resources/software_repositories/allowlists`
```

```

In this check list we aim to evaluate our deployment against the {ref}`security configuration <design_turing_security_configuration>` that we apply at the Alan Turing Institute.
The security checklist currently focuses on checks that can evaluate these security requirements for {ref}`policy_tier_2` (or greater) SREs (with some steps noted as specific to a tier):

## How to use this checklist

Ensure you have met the [](#prerequisites).
Work your way through the actions described in each section, taking care to notice each time you see a {{camera}} or a {{white_check_mark}} and the word Verify.

```{note}
- {{camera}} Where you see the camera icon, there should be accompanying screenshot(s) of evidence for this item in the checklist (you may wish to save your own equivalent screenshots as evidence)
- {{white_check_mark}} This indicates a checklist item for which a screenshot is either not appropriate or difficult
```

## Prerequisites

### Roles

The following roles will be needed for this checklist

> - Reseearch User who is a member of the SRE A research users group

Reviewer 2: ```suggestion
- Research User who is a member of the SRE A research users group
```

>     - Ensure this person's IP address is added to the `reserach_user_ip_addresses` list in SRE A's configuration.

Reviewer 2: ```suggestion
    - Ensure this person's IP address is added to the `research_user_ip_addresses` list in SRE A's configuration.
```

{{camera}} <b>Verify that:</b>

<details><summary>MFA setup is successful</summary>

```{image} security_checklist/aad_additional_security_verification.png
:alt: AAD additional security verification
:align: center
```
</details>
````

#### User can login after setting up MFA

Check that the research user can authenticate using MFA and is granted access to the SRE.

- Login to the remote desktop web client as the research user.

````{attention}
{{camera}} <b>Verify that:</b>
<details><summary>you are prompted for MFA and can respond</summary>

```{image} security_checklist/aad_mfa_approve_signin_request.png
:alt: AAD MFA approve sign-in request
:align: center
```

</details>
````

#### Authenticated user can access workspaces

Check that the research user can access a workspace.

- Login to the remote desktop web client as the research user.
- Select a workspace and login as the research user.

````{attention}
{{camera}} <b>Verify that:</b>
<details><summary>you can connect to <i>Desktop: Ubuntu0</i></summary>

```{image} security_checklist/guacamole_srd_desktop.png
:alt: SRD desktop
:align: center
```

</details>
````

## 2. Isolated Network

### Turing configuration setting:

> - {ref}`Researchers <role_researcher>` cannot access any part of an SRE (except the portal) from outside the SRE network.

Reviewer 2: Does this mean the Guacamole dashboard? Or the Azure portal? I don't think either of these should be accessible from outside an approved network.

Author: The Guacamole dashboard. "outside the SRE network" means "outside of the SRE vnet".

Or, another way of saying it would be "The Guacamole dashboard is the only part of the SRE exposed to the internet."

Author: Updated in 06b0938
